### PR TITLE
Date and times in R curriculum

### PR DESCRIPTION
**Pull request by jessicaguo**
Outline of notes and code for code along. Needs review and then a dry run to see if times are realistic.

### CODE REVIEW

File: 202204-datestimesR/notes.Rmd [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+x2 <- force_tz(x, tzone = "America/Phoenix")
+x; x2
+difftime(x, x2)
+```
+
+## Help
+
+Feel free to drop by the [CCT Data Science Team office
+hours](https://datascience.cals.arizona.edu/office-hours), which happen
```
Comment (dlebauer) [inline 202204-datestimesR/notes.Rmd:267]:
```suggestion
hours](https://datascience.cals.arizona.edu/office-hours), which happens
```

**Review by dlebauer — COMMENTED** [review]
I think this looks like a comprehensive overview of date times. Youve done a good job of identifying key concepts. If you weren't already planning to do so, if there is time, it fits, I have a few thoughts.

The main suggestion is to give an introduction overall and then to each section. Perhaps introduce the scope and logic of the order of topics, how they fit together, if there is an underlying logic to the order. When you get to lubridate, explain what it does and does not do, when to use base R and when to use lubridate.

Another way to introduce could be to mention motivating problems, in what contexts do you find this type of problem, where is this type of operation required, what problem does this feature solve? 

- to motivate 'why do I need to know the difference between POSIXct POSIXlt? If I need to know, what is the theory? I just ran into `ERROR: Invalid Input: objects of class POSIXct`? why! (maybe google it or search on stackexchange)
- time differences are useful for converting states to rates? diff(biomass)/diff(t) = growth rate. what else are they useful for? time between events, like number of days since planting or frequency of rainfall events
- when will I have problems with time zones? eg weather is in GMT, photosynthesis is in local time. This is just a synthetic dataset I but could get the point across; could start with 'what is wrong with this picture?'
image

File: 202204-datestimesR/notes.Rmd [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+z <- ISOdatetime(year = 2021, month = 2, day = 23, hour = 20, min = 30, sec = 0,
+                 tz = "America/Phoenix")
+z
+
+# Try converting the following: 
+"06:59 Sep 02 1995"
+"16/Oct/2005:07:51:00"
+"12-Mar-2012 0830"
+2013; 8; 18
```
Comment (KristinaRiemer) [inline 202204-datestimesR/notes.Rmd:126]: What is the purpose of this example? Is it to add a time where there isn't yet one?

**Review by KristinaRiemer — APPROVED** [review]
I think this looks great! I agree with David about providing some motivating reasons for using the different tools. Based on feedback from previous workshop participants, pointing at examples in your own work is a great way to do that!

File: 202204-datestimesR/notes.Rmd (hunk 2 above)
Comment (jessicaguo) [reply 202204-datestimesR/notes.Rmd:132]: These are intended to be hands-on exercises. The last case is if the date is provided as separate numeric values of year, month, and day.

**Pull request merged**